# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/glibc23/regcomp.c.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---


This fixes grep -E  -q '^[^#]*\bflistdel

  - Rene Rebe <rene@rocklinux.org>

--- glibc-2.3.3-2003-12-18/posix/regcomp.c	2003-12-16 07:10:51.000000000 +0100
+++ glibc-2.3.3-2003-12-18-fixed/posix/regcomp.c	2003-12-21 01:26:01.000000000 +0100
@@ -1575,7 +1575,8 @@
 		? dfa->nodes[node].opr.ctx_type : 0);
   /* If the current node has constraints, duplicate all nodes.
      Since they must inherit the constraints.  */
-  if (constraint && !dfa->nodes[dfa->edests[node].elems[0]].duplicated)
+  if (constraint && dfa->edests[node].nelem &&
+      !dfa->nodes[dfa->edests[node].elems[0]].duplicated)
     {
       int org_node, cur_node;
       org_node = cur_node = node;
